src/vtab/mod.rs

```diff
@@ -331,6 +332,21 @@ impl IndexInfo {
         }
     }
 
+    /// String passed if/when filter() is called on the cursor
+    pub fn set_idx_str(&mut self, idx_str: &str) {
+        unsafe {
+            let c_idx_str = CString::new(idx_str).
+                expect("Couldn't create C string (probably because idx_str contains a null (\\0) character)").
+                into_bytes_with_nul();
+
+            let dst = ffi::sqlite3_malloc(c_idx_str.len() as i32) as *mut ::std::os::raw::c_char;
```

Comment:
See [alloc](https://github.com/rusqlite/rusqlite/blob/bb34f3fec8a8ed71d203a452a147cd2d58348748/src/vtab/mod.rs#L1039) and `SqliteMallocString`